Assets/Scripts/Animals/AnimalController.cs

```diff
@@ -55,36 +64,59 @@ private void OnTriggerExit(Collider other)
         }
     }
 
+    public void Scared(Vector3 playerPos)
+    {
+        //set pos to the other side of the planet from the player. This doesn't work particularly well but it works for now
+        pos = playerPos * -1;
+        LookAtMovement();
+    }
+
+    public void Hostile(Vector3 playerPos)
+    {
+        pos = playerPos;
+        LookAtMovement();
+        attacking = true;
+    }
+
     IEnumerator wait()
     {
         while (true)
         {
-            Vector3 pos = Random.onUnitSphere * radius; //picks a random point on the surface of a sphere with the radius
-            transform.LookAt(pos, transform.position * Time.deltaTime); //makes them face the direction they will move to
-
-            //https://forum.unity.com/threads/help-using-coroutine-to-move-game-object-to-position-wait-then-return-to-original-position.1122784/
-            //BLESSED UNITY FORUMS
-            RaycastHit hit;
-
-            while (Vector3.Distance(transform.position, pos) > 1) //while the animal is not at their desired position
-            { //This code is a mess but it works and every time I try to make it better it stops working.
-                if (
-                    Physics.Raycast(transform.position, pos, out hit, .5f) //if there is a collider in front of the animal
-                    && hit.collider.gameObject.tag == "OnPlanetCollision" //if the collider is for collision
-                    && hit.collider.gameObject.transform.parent.gameObject != gameObject //if the collider is not the animal's own collider
-                )
-                {
-                    //end the movement
-                    break;
-                }
-                transform.position = Vector3.MoveTowards(
-                    transform.position,
-                    pos,
-                    SPEED * Time.deltaTime
-                ); //move them to it!
-                yield return 0; //used to let the engine wait for a frame which breaks an endless broken loop
+            if (!moving)
+            {
+                pos = Random.onUnitSphere * radius; //picks a random point on the surface of a sphere with the radius
+                LookAtMovement();
             }
             yield return new WaitForSeconds(Random.Range(MINWAIT, MAXWAIT)); //pause for a random time and then go again
         }
     }
+
+    //Moving towards the current position
+    private void Movement()
+    {
+        moving = false;
+        if (
+            Vector3.Distance(transform.position, pos) > 2 //if the animal is not at the desired position
+            && //If any of the following three statements are false it returns true. If all are true it returns false. This is a manual way of making a NAND gate
+            (
+                !Physics.Raycast(transform.position, pos, out RaycastHit hit, .5f)
```

Comment:
Also move this up a line. I think it's better formatting when you drop down a line at a condition (II,&& etc) rather than letting parenthesis be open and alone. They are not curly braces and shouldn't act like it!

Reply:
This is how the agreed upon code formatter, csharpier, formats code by default.